langchain/src/chat_models/base.ts

```diff
@@ -0,0 +1,864 @@
+  }
+}
+
+interface ConfigurableModelFields {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  defaultConfig?: Record<string, any>;
+  /**
+   * @default "any"
+   */
+  configurableFields?: string[] | "any";
+  /**
+   * @default ""
+   */
+  configPrefix?: string;
+  /**
+   * Methods which should be called after the model is initialized.
+   * The key will be the method name, and the value will be the arguments.
+   */
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  queuedMethodOperations?: Record<string, any>;
+}
+
+class _ConfigurableModel<
+  RunInput extends BaseLanguageModelInput = BaseLanguageModelInput,
+  CallOptions extends ConfigurableChatModelCallOptions = ConfigurableChatModelCallOptions
+> extends Runnable<RunInput, AIMessageChunk, CallOptions> {
+  lc_namespace = ["langchain", "chat_models"];
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  _defaultConfig?: Record<string, any> = {};
+
+  /**
+   * @default "any"
+   */
+  _configurableFields: string[] | "any" = "any";
+
+  /**
+   * @default ""
+   */
+  _configPrefix: string;
+
+  /**
+   * Methods which should be called after the model is initialized.
+   * The key will be the method name, and the value will be the arguments.
+   */
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  _queuedMethodOperations: Record<string, any> = {};
+
+  constructor(fields: ConfigurableModelFields) {
+    super();
+    this._defaultConfig = fields.defaultConfig ?? {};
+
+    if (fields.configurableFields === "any") {
+      this._configurableFields = "any";
+    } else {
+      this._configurableFields = fields.configurableFields ?? "any";
+    }
+
+    if (fields.configPrefix) {
+      this._configPrefix = fields.configPrefix.endsWith("_")
+        ? fields.configPrefix
+        : `${fields.configPrefix}_`;
+    } else {
+      this._configPrefix = "";
+    }
+
+    this._queuedMethodOperations =
+      fields.queuedMethodOperations ?? this._queuedMethodOperations;
+  }
+
+  async _model(config?: RunnableConfig) {
+    const params = { ...this._defaultConfig, ...this._modelParams(config) };
+    let initializedModel = await _initChatModelHelper(
+      params.model,
+      params.modelProvider,
+      params
+    );
+
+    // Apply queued method operations
+    const queuedMethodOperationsEntries = Object.entries(
+      this._queuedMethodOperations
+    );
+    if (queuedMethodOperationsEntries.length > 0) {
+      for (const [method, args] of queuedMethodOperationsEntries) {
+        if (
+          method in initializedModel &&
+          // eslint-disable-next-line @typescript-eslint/no-explicit-any
+          typeof (initializedModel as any)[method] === "function"
+        ) {
+          // eslint-disable-next-line @typescript-eslint/no-explicit-any
+          initializedModel = await (initializedModel as any)[method](...args);
+        }
+      }
+    }
+
+    return initializedModel;
+  }
+
+  bindTools(
```

Comment:
`override`

Reply:
This class extends `Runnable` so I don't think that is necessary

Reply:
Now we need it lol